### DISCUSSION TITLE
Import external libraries in python scripts

### DISCUSSION THREAD

**Discussion by rafinhafpb** [bonsai-rx/bonsai#1926] [Q&A]
Hi Bonsai community!

I was wondering if it's possible to import external libraries into my `PythonTransform` node. The operation I would like to perform involves reading a .bin file created by a `MatrixWriter` node, using the NumPy .reshape function, and then writing the data into a .txt file with a specific format.

I have seen previous discussions in the Google discussion group about doing this by relying on IronPython external calls and the import of some DLLs, but I had no success. Therefore, I wanted to know if there is an easier way to do it now. I also noticed some nodes such as `CreateModule` or `CreateRuntime` that might be used to import modules, but I wasn't able to figure out how they work.

I appreciate any kind of help!

**Comment by bruno-f-cruz** [comment]
IronPython will not make your life easy if you are trying to import external libraries, especially numpy which has a fair amount of dependencies.

The newer python scripting package might be what you are looking for as you pointed out.
I would probably start with the documentation of the package: https://bonsai-rx.org/python-scripting/index.html?tabs=windows

**Reply by rafinhafpb** [reply]
Hi Bruno, thank you for your answer. 
I followed the instructions in the documentation, but when I try to start Bonsai, I get the error message shown below. I saw that someone had already asked about this error in the forum, but what worked for them didn't work for me: changing Python's version to 3.10 instead of 3.12. Do you have any ideas on what it could be?
![image](https://github.com/user-attachments/assets/32c3d3b9-5ec0-4883-b91f-1f13c4e929d5)

**Reply by bruno-f-cruz** [reply]
This error usually gets thrown when your python kernel cannot be imported. I have not come across it in a while (mostly python version problems before) but maybe @glopesdev and @ncguilbeault have a better idea since they have been using it recently.
